publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -96,6 +99,16 @@ public Publication createPublicationWithPredefinedCreationDate(Publication input
         return insertResource(newResource);
     }
 
+    public Publication createPublicationWhilePersistingEntryFromLegacySystems(Publication inputData)
+            throws TransactionFailedException {
+        Resource newResource = Resource.fromPublication(inputData);
```

Comment:
We have created a weird situation where we allow a publication to be DRAFT, i.e., NOT PUBLISHED but still to have a published date. Except for the fact that this does not stand, this date will disappear when the Publication is actually Published. Since all Publications from legacy systems are already published, I suggest we just set that status of all Publications to PUBSLISHED when this method is  called. That is to say it is the Service and this specific method that will set the publication status to PUBLISHED so that a situtaion where a Publication is not published but still has a publication date is not possible.